pkg/cmd/pulumi/new.go

```diff
@@ -183,26 +191,28 @@ func runNew(ctx context.Context, args newArgs) error {
 
 		// Set the org name for future use.
 		orgName = parts[0]
-
 		projectName := parts[1]
-		if err := validateProjectName(ctx, b, orgName, projectName, args.generateOnly, opts); err != nil {
-			return err
-		}
 
 		stackName, err := buildStackName(args.stack)
 		if err != nil {
 			return err
 		}
+
 		existingStack, existingName, existingDesc, err := getStack(ctx, b, stackName, opts)
 		if err != nil {
 			return err
 		}
-		s = existingStack
-		if args.name == "" {
-			args.name = existingName
-		}
-		if args.description == "" {
-			args.description = existingDesc
+		if existingStack != nil {
+			s = existingStack
+			if args.name == "" {
+				args.name = existingName
+			}
+			if args.description == "" {
+				args.description = existingDesc
+			}
+		} else {
```

Comment:
I made this change as we previously did not check if the stack was nil. We need to try to set `args.name` so that we don't prompt for the project name as we have it from the stack reference.

Reply:
I think we could simplify getStack to not return the project at all and just unconditionally set args.name from the projectName pulled from parts above.

Reply:
It'd be surprising if existingStack's project name != project (from `-s org/project/stack`) so I'll make the change.